src/tools/python/python.ts

```diff
@@ -74,34 +67,18 @@ export class PythonTool extends Tool<PythonToolOutput, PythonToolOptions> {
   ].join(" ");
 
   public readonly storage: PythonStorage;
+  protected files: PythonFile[] = [];
 
   async inputSchema() {
-    const files = await this.storage.list();
-    const fileIds = unique(map(files, ({ id }) => id));
-
-    const zodFileId = hasMinLength(files, 2)
-      ? z.union(map(files, (file) => z.literal(file.id).describe(file.filename)))
-      : hasMinLength(files, 1)
-        ? z.literal(files[0].id).describe(files[0].filename)
-        : z.undefined();
-
+    this.files = await this.storage.list();
```

Comment:
I think that storing results in `this.files` is unnecessary.

Reply:
You can enable caching on the `inputSchema` method via the `@Cache` decorator, which you then revalidate in the `_run` method. 

See: https://i-am-bee.github.io/bee-agent-framework/#/cache?id=cache-decorator-cache (Complex Example)

Reply:
I need the `id`s that are deliberately _not_ part of the `inputSchema`, thus I need to cache the full `storage.list()` result. Making a new function with `@Cache` only to invalidate it would just complicate the code -- how is `@Cache({ cacheKey: SingletonCacheKeyFn })` and `Cache.getInstance(this, "id").clear()` clearer over simply using an instance variable?